tests/zfs-tests/tests/functional/userquota/groupspace_003_pos.ksh

```diff
@@ -0,0 +1,103 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
+# Use is subject to license terms.
+#
+
+#
+# Copyright (c) 2013 by Delphix. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/tests/functional/userquota/userquota_common.kshlib
+
+#
+# DESCRIPTION:
+#       Check the user used and groupspace size in zfs groupspace
+#
+#
+# STRATEGY:
+#       Create objects for different users in the same group and verify object
+#	count
+#
+
+function cleanup
+{
+	if datasetexists $snapfs; then
+		log_must $ZFS destroy $snapfs
+	fi
+	log_must cleanup_quota
+}
+
+function group_object_count
+{
+	typeset fs=$1
+	typeset user=$2
+	typeset cnt=$($ZFS groupspace -oname,objused $fs | $GREP $user |
+			$AWK '{print $2}')
+	echo $cnt
+}
+
+function sync_pool
+{
+	$ZPOOL scrub $TESTPOOL
+}
```

Comment:
The same function already exists in `functional/redundancy/redundancy.kshlib` you could move it to `libtest.shlib` so it's available to all the test cases.